lambdas/data-migration2/tests/test-granulesAndFiles.js

```diff
@@ -781,6 +880,139 @@ test.serial('queryAndMigrateGranuleDynamoRecords only processes records for spec
   });
 });
 
+test.serial('migrateGranulesAndFiles throws if migrateAndOverwrite and migrateOnlyFiles are set to true', async (t) => {
+  const {
+    knex,
+    testGranule,
+    testExecution,
+    testCollection,
+  } = t.context;
+
+  const testGranule1 = testGranule;
+  const testGranule2 = generateTestGranule({
+    collectionId: constructCollectionId(testCollection.name, testCollection.version),
+    execution: testExecution.url,
+  });
+
+  await Promise.all([
+    granulesModel.create(testGranule1),
+    granulesModel.create(testGranule2),
+  ]);
+
+  t.teardown(async () => {
+    await Promise.all([
+      granulesModel.delete({ granuleId: testGranule.granuleId }),
+      granulesModel.delete({ granuleId: testGranule2.granuleId }),
+    ]);
+  });
+
+  await t.throwsAsync(
+    migrateGranulesAndFiles(process.env, knex, {
+      migrateAndOverwrite: 'true',
+      migrateOnlyFiles: 'true',
+    }),
+    { instanceOf: InvalidArgument }
+  );
+});
+
+test.serial('migrateGranulesAndFiles reports failures if GranuleMigrationParams.migrateOnlyFiles is set to true and the granules were not previous', async (t) => {
```

Comment:
I don't totally understand the wording here. Did you mean "and the granules were not previously...."?

Reply:
😅  oops, editor window sizes strike again.   Yes, that's truncated. 

b91d3401f2